modal_proto/api.proto

```diff
@@ -1147,6 +1164,25 @@ message FunctionOptions {
   bool replace_secret_ids = 12;
 }
 
+
+message FunctionParameter {
+  string name = 1;
+  FunctionParameterType type = 2;
+}
```

Comment:
I am just continuing to be annoying about this, but the terminology here remains really confusing — "FunctionParameter" is actually a constructor parameter for a modal Cls, not one of the parameters of a modal Function, yeow.

Reply:
Yeah I can totally see how this is confusing 😬 We could rename this ClassParameter if that makes more sense? In Modal terminology I guess the parameters of the function would be "arguments".

I think I've intermixed class_parameter and function_parameter a bit in naming as a consequence of it being called `serialized_params` and being an attribute of a `Function`, although the (backend) Function syntactically is exposed as a `class` in the SDK.

Reply:
Did a rename pass on the PR now:
`FunctionParameter -> ClassParameterSpec`   (making it clearer this is part of the schema and not a value of a parameter)
`FunctionParameterSet -> ClassParameterSet`
`FunctionParameterValue -> ClassParameterValue`

Reply:
Made another pass to bundle up the `repeated ClassParameterSpec` + ParameterSerializationFormat into a `ClassParameterInfo` since they are always used together (good that proto allows us to rename thihngs later...)